book/configuration.rst

```diff
@@ -1,15 +1,55 @@
 .. index::
    single: Configuration
+   single: Environments
 
-Configuring Symfony (and Environments)
-======================================
+Configuring Symfony and Environments
+====================================
+
+.. index::
+   single: Configuration; Formats
+
+Configuration Formats
+---------------------
+
+Symfony supports four configuration formats: YAML, XML and PHP. Throughout
+the chapters, all configuration examples are shown in all four formats. Each
+has its own advantages and disadvantages. The choice of which to use is up
+to you:
+
+* *YAML*: Simple, clean and readable (learn more about YAML in the Yaml
+  component documentation :doc:`/components/yaml/yaml_format`);
+
+* *XML*: More powerful than YAML at times and supports IDE autocompletion;
+
+* *PHP*: Very powerful but less readable than standard configuration formats.
+
+* *Annoatations*: Code and configuration are in the same place, simple to
+  learn and to use, concise to write.
+
+The supported formats with their orders are:
+
+* Configuration (including services): YAML, XML, PHP
+
+* Routing: Annotations, YAML, XML, PHP
+
+* Validation: Annotations, YAML, XML, PHP
+
+* Doctrine Mapping: Annotations, YAML, XML, PHP
+
+* Translation: XML, YAML, PHP
```

Comment:
Since this is a Configuration chapter i thought it would be convenient to include this information. Actually in my humble personal opinion this information should be present in some earlier chapter of the book. Since this part of the documentation is called "the book" it would be logical to include this before the first examples and encounter of different formats - some programmers who might be just learning about Symfony could decide to read this "book" as a book - from 1st chapter to the last - and it would make sense to have this information at the beginning.

@WouterJ, @stof could i get your opinion on this? Thank you.